.github/workflows/filecoin-pin-build.yml

```diff
@@ -0,0 +1,34 @@
+name: Build Site
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+  actions: write  # needed to upload artifacts
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build site
+        run: npm run build
```

Comment:
I was surprised that there wasn't existing CI to hook into, but I guess that is abstracted away by a fleek app or something else